config/githooks/pre-push

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+cd $(git rev-parse --show-toplevel)
+
+# copyright check
+copyright=true
+if git grep -l "^\(#\|//\).*\(\-2020\|\ 2010,\)" > matches.txt
+then
+  echo "Please update the following files to Copyright (c) 2010-2021:"
+  cat matches.txt
+  copyright=false
+fi
+
+# license check
+license=true
+if git grep -li "^\(#\|//\).*GNU\ Lesser\ General\ Public\ License" > matches.txt
+then
+  echo "Please update the following files to the BSD-3 license:"
+  cat matches.txt
+  license=false
+fi
+
+# release check
+release=true
+if git grep -l "^\(#\|//\).*LLNL\-CODE\-443211" > matches.txt
+then
+  echo "Please update the following files to LLNL-CODE-806117:"
+  cat matches.txt
+  release=false
+fi
+
+# wrap-up
+code=0
+if ! $copyright ; then
+  echo "copyright check failed, unroll log for details"
+  code=1
+fi
+if ! $license ; then
+  echo "license check failed, unroll log for details"
+  code=1
+fi
+if ! $release ; then
+  echo "release check failed, unroll log for details"
+  code=1
+fi
+
+## style check
```

Comment:
Why is it commented out?

Reply:
That’s because `./runtest code-style` is not just a check. It will actually reformat the code.

I figured some could be mad that a push hook rejected the push and changed the code. At least for a first hook experience.

In fact, one can push while in a dirty state, in which case format changes would get mixed with pending changes...

What do you think?

Reply:
I think that decision makes sense, but maybe a comment with a short sentence explaining why it's commented would be useful to people reading the code

Reply:
For sure, it took me a while, and some tests to remember why myself...